dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -399,6 +400,9 @@ private boolean shouldAttemptToRetry(Throwable throwable) {
                             // String matches CJR RemotingOkHttpCall.shouldRetry
                             && socketTimeout.getMessage().contains("connect timed out");
                 }
+                if (isEtimedoutException(throwable)) {
```

Comment:
minor nit: it might be worth a note explaining that the native `read` implementation sometimes throws `SocketException` with this specific text and that that is distinct from `SocketTimeoutException`.

otherwise lgtm